Projects/ChangeTier/Sources/Views/CompareTierScreen.swift

```diff
@@ -22,12 +22,12 @@ struct CompareTierScreen: View {
             ScrollableSegmentedView(
                 vm: scrollableSegmentedViewModel,
                 contentFor: { id in
-                    let tier = vm.tiers.first(where: { $0.id == id })
-                    let limits = tier?.productVariant?.insurableLimits ?? []
-                    let perils = tier?.productVariant?.perils ?? []
+                    let currentTier = vm.tiers.first(where: { $0.id == id })
+                    let currentLimits = currentTier?.productVariant?.insurableLimits ?? []
+                    let perils = getFilteredPerils(currentTier: currentTier)
```

Comment:
It would be nice to have perils already in place instead of calling this function every time.

Reply:
moved it to init